src/AttributeHandling/ParameterAttributeResolveContext.php

```diff
@@ -23,6 +24,7 @@ public function __construct(
         private ReflectionParameter|ReflectionProperty $parameter,
         private Result $resolveResult,
         private DataInterface $data,
+        private HydratorInterface $hydrator,
```

Comment:
```suggestion
        private ?HydratorInterface $hydrator = null,
```

And throw `LogicException` in `getHydrator()` when hydrator is not set. It's need for BC. We make hydrator required in next major version.